kernel/kernel-benchmarks/src/test/java/io/delta/kernel/benchmarks/workloadrunners/WriteRunner.java

```diff
@@ -144,34 +143,48 @@ public void executeAsBenchmark(Blackhole blackhole) throws Exception {
   /** Cleans up the state created during benchmark execution by reverting all committed changes. */
   @Override
   public void cleanup() throws Exception {
-    if (initialDeltaLogFiles == null) {
-      return; // Setup didn't complete, nothing to clean up
+    if (!initialDeltaLogFiles.isPresent()) {
+      throw new RuntimeException("Cannot cleanup before setup is called.");
     }
     // Delete any files that weren't present initially
     Set<String> currentFiles = captureFileListing();
+    Set<String> initialFiles =
+        initialDeltaLogFiles.orElseThrow(
+            () -> new RuntimeException("Cannot cleanup before setup is called."));
     for (String filePath : currentFiles) {
-      if (!initialDeltaLogFiles.contains(filePath)) {
+      if (!initialFiles.contains(filePath)) {
         engine.getFileSystemClient().delete(filePath);
       }
     }
   }
 
-  /** @return a set of all file paths in the `_delta_log/` directory of the table. */
+  /**
+   * @return all file paths in the `_delta_log/` directory (including `_staged_commits/` if
+   *     CatalogManaged)
+   */
   private Set<String> captureFileListing() throws IOException {
-    // Construct path prefix for all files in `_delta_log/`. The prefix is for file with name `0`
-    // because the filesystem client lists all _sibling_ files in the directory with a path greater
-    // than `0`.
-    String deltaLogPathPrefix =
-        new Path(workloadSpec.getTableInfo().getResolvedTableRoot(), "_delta_log/0")
-            .toUri()
-            .getPath();
+    List<String> prefixes = new ArrayList<>(Collections.singletonList("_delta_log"));
+    if (workloadSpec.getTableInfo().isCatalogManaged()) {
+      prefixes.add("_delta_log/_staged_commits");
+    }
 
     Set<String> files = new HashSet<>();
-    try (CloseableIterator<FileStatus> filesIter =
-        engine.getFileSystemClient().listFrom(deltaLogPathPrefix)) {
-      while (filesIter.hasNext()) {
-        FileStatus file = filesIter.next();
-        files.add(file.getPath());
+    for (String prefix : prefixes) {
+      // Construct path prefix for all files in `_delta_log/`. The prefix is for file with name `0`
+      // because the filesystem client lists all _sibling_ files in the directory with a path
+      // greater than `0`.
+      String deltaLogPathPrefix =
+          new Path(workloadSpec.getTableInfo().getResolvedTableRoot(), new Path(prefix, "0"))
+              .toUri()
+              .getPath();
+
+      // List from the lowest version in the prefix
+      try (CloseableIterator<FileStatus> filesIter =
+          engine.getFileSystemClient().listFrom(deltaLogPathPrefix)) {
+        while (filesIter.hasNext()) {
+          FileStatus file = filesIter.next();
```

Comment:
nit: no need to define `file` as its own variable when it is just being used to access `getPath()` -- this can all be 1 line